src/mon/OSDMonitor.cc

```diff
@@ -293,7 +294,21 @@ void OSDMonitor::update_from_paxos(bool *need_bootstrap)
   /** we don't have any of the feature bit infrastructure in place for
    * supporting primary_temp mappings without breaking old clients/OSDs.*/
   assert(g_conf->mon_osd_allow_primary_temp || osdmap.primary_temp->empty());
-
+  {
+    bufferlist bl;
+    mon->store->get(OSD_PG_CREATINGS_PREFIX, "creatings", bl);
+    auto p = bl.begin();
+    creating_pgs.decode(p);
+    for (const auto& pg : creating_pgs.pgs) {
+      int acting_primary = -1;
+      auto pgid = pg.first;
+      auto created = pg.second.first;
+      mapping.get(pgid, nullptr, nullptr, nullptr, &acting_primary);
```

Comment:
key is probably just to make sure the MOSDPGCreate message is tagged with the epoch the mapping was calcualted and not the current paxos epoch (since they might vary?)